app/src/main/java/com/example/hris/ui/viewmodels/profile/UpdateProfileViewModel.kt

```diff
@@ -18,27 +21,76 @@ class UpdateProfileViewModel @Inject constructor(
     val userData = hrisRepository.userData
     val loadingDialogState = MutableLiveData<Boolean>()
     val message = MutableLiveData<String>()
+    val isSuccessfull = MutableLiveData<Unit>()
 
     fun updateProfile(
         firstName: String,
         middleName: String?,
         lastName: String,
         emailAddress: String,
         mobileNumber: String,
-        landline: String?
+        landline: String?,
+        context: Context
     ) {
-        viewModelScope.launch {
-            update(firstName, middleName, lastName, emailAddress, mobileNumber, landline)
+        if (!nameIsValid(firstName)) {
+            message.value = context.getString(R.string.first_name_error)
+        } else if (!middleNameIsValid(middleName)) {
+            message.value = context.getString(R.string.middle_name_error)
+        } else if (!nameIsValid(lastName)) {
+            message.value = context.getString(R.string.last_name_error)
+        } else if (!emailIsValid(emailAddress)) {
+            message.value = context.getString(R.string.email_address_error)
+        } else if (!mobileNumberIsValid(mobileNumber)) {
+            message.value = context.getString(R.string.mobile_number_error)
+        } else if (!landlineNumberIsValid(landline)) {
+            message.value = context.getString(R.string.landline_number_error)
+        } else {
```

Comment:
Move the validation block to another method. Like so:
![image](https://user-images.githubusercontent.com/56427591/213640575-e86c0791-e224-4572-97e6-51787629f674.png)